src/primitives.jl

```diff
@@ -179,6 +181,30 @@ end
 dropnull(X::NullableVector) = X.values[!X.isnull]    # -> Vector
 
 """
+    dropnull!(X::AbstractVector)
+
+Remove null entries of `X` in-place and return the updated vector. This is a
+no-op if no nulls are present.
+
+    dropnull!(X::NullableVector)
+
+Remove null entries of `X` in-place and return a `Vector` view of `X.values`
+"""
+function dropnull!(X::AbstractVector)
+    if !(Nullable <: eltype(X))
+        return X
+    end
+    deleteat!(X, find(isnull, X))
```

Comment:
What I meant is that `isa` is evaluated at compile time if possible. So you can have the `isa` call inside the loop for all cases, and it won't be called when `T <: Nullable`. Of course for `T = Any` there is no choice but calling it on each element.

The `dropnull` methods can also be merged indeed.